js/models/Profile.js

```diff
@@ -0,0 +1,158 @@
+import _ from 'underscore';
+import BaseModel from './BaseModel';
+import app from '../app';
+import is from 'is_js';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      primaryColor: '#086A9E',
+      secondaryColor: '#317DB8',
+      textColor: '#ffffff',
+      social: [],
+    };
+  }
+
+  url() {
+    // url is handled by sync, but backbone bombs if I don't have
+    // something explicitly set
+    return 'use-sync';
+  }
+
+  getColorFields() {
+    return [
+      'primaryColor',
+      'secondaryColor',
+      'textColor',
+    ];
+  }
+
+  // Ensure any colors are strings and have a leading hash.
+  standardizeColorFields(attrs = {}) {
+    const updatedAttrs = { ...attrs };
+
+    this.getColorFields().forEach((field) => {
+      if (typeof attrs[field] !== 'undefined') {
+        updatedAttrs[field] = updatedAttrs[field].toString();
+        updatedAttrs[field] = updatedAttrs[field].charAt(0) !== '#' ?
+          `#${updatedAttrs[field]}` : updatedAttrs[field];
+      }
+    });
+
+    return updatedAttrs;
+  }
+
+  set(key, val, options) {
+    // Handle both `"key", value` and `{key: value}` -style arguments.
+    let attrs;
+    let opts = options;
+
+    if (typeof key === 'object') {
+      attrs = key;
+      opts = val;
+    } else {
+      (attrs = {})[key] = val;
+    }
+
+    if (attrs.social) {
+      if (!_.isEqual(attrs.social, this._social)) {
+        this.attributes.social = this._social;
+
+        // if the object contents are not equal, we'll clone a new object so
+        // the change is event is fired.
+        this._social = JSON.parse(JSON.stringify(attrs.social)); // deep clone
+        attrs.social = JSON.parse(JSON.stringify(attrs.social));
```

Comment:
No, it's critical that the two variables reference completely different objects (on a deep level) and that wouldn't happen if we only "deep cloned" once. They'd both reference the same clone.